contrib/sssd.spec.in

```diff
@@ -741,91 +446,96 @@ UIDs/GIDs to names and vice versa. It can be also used for mapping principal
 
 %package -n libsss_certmap
 Summary: SSSD Certificate Mapping Library
-Group: Development/Libraries
 License: LGPLv3+
-Requires(post): /sbin/ldconfig
-Requires(postun): /sbin/ldconfig
+Conflicts: sssd-common < %{version}-%{release}
 
 %description -n libsss_certmap
 Library to map certificates to users based on rules
 
 %package -n libsss_certmap-devel
 Summary: SSSD Certificate Mapping Library
-Group: Development/Libraries
 License: LGPLv3+
 Requires: libsss_certmap = %{version}-%{release}
 
 %description -n libsss_certmap-devel
 Library to map certificates to users based on rules
 
-%if (0%{?with_kcm} == 1)
 %package kcm
 Summary: An implementation of a Kerberos KCM server
-Group:  Applications/System
 License: GPLv3+
 Requires: sssd-common = %{version}-%{release}
 %{?systemd_requires}
 
 %description kcm
 An implementation of a Kerberos KCM server. Use this package if you want to
 use the KCM: Kerberos credentials cache.
-%endif
 
 %prep
-%setup -q -n %{name}-%{version}
+# Update timestamps on the files touched by a patch, to avoid non-equal
+# .pyc/.pyo files across the multilib peers within a build, where "Level"
+# is the patch prefix option (e.g. -p1)
+# Taken from specfile for python-simplejson
+UpdateTimestamps() {
+  Level=$1
+  PatchFile=$2
+
+  # Locate the affected files:
+  for f in $(diffstat $Level -l $PatchFile); do
+    # Set the files to have the same timestamp as that of the patch:
+    touch -r $PatchFile $f
+  done
+}
+
+%setup -q
+
+for p in %patches ; do
```

Comment:
Would it be possible to use `%autosetup` and is it important to have `UpdateTimestamps`?

Reply:
There is a comment about 'UpdateTimestamps' but I hardly understand it: "to avoid non-equal .pyc/.pyo files across the multilib peers within a build"...

Btw, it seems this function is the only reason for 'BuildRequires: diffstat'

Reply:
I was not aware of %autosetup, looks good.

UpdateTimestamps is used to fix some rpmdiff issues that would have to be waved on each release.

Reply:
UpdateTimestamps is used to suppress multilib regression warning in rpmdiff that needs to be waived otherwise. But I checked some packages (including the one mentioned in comments) and they all use %autosetup now and there is no mention of this kind of function, though I was not able to find where it disappeared. Let's try to build a scratch build for rhel and run rpmdiff there to see if it is still needed or not.